### PR TITLE
Add Lightweight Runtime Benchmarking for Loss Functions

### PR DESCRIPTION
**Pull request by s1k0ra**
This PR adds a script for a lightweight benchmark for the runtime of:

- `CrossEntropyLoss`
- `NumberTokenLoss`
- `CEWithNTL`

under two modes for the NTL calculation (to avoid NaNs):
- `tokenizer_modified`: all tokens treated as number tokens  
- `numbers_only`: only number tokens used in labels tensor

Key features:
- Uses only lightweight NTL version
- Accurate timing with CUDA sync
- Optional token filtering to avoid NaNs for NTL
- Logs mean ± std for each loss over multiple batches

**Sample output:**
```
Tokenzer modified (All tokens encoded as number tokens for ntl loss)
CrossEntropyLoss: (32.52 ± 0.50)ms
NumberTokenLoss: (9.15 ± 0.15)ms
CEWithNTL: (41.15 ± 0.29)ms
    
Numbers only (Only number tokens are used to compute ntl loss)
CrossEntropyLoss: (33.35 ± 0.02)ms
NumberTokenLoss: (0.44 ± 0.01)ms
CEWithNTL: (33.67 ± 0.02)ms
```

Useful for lightweight runtime comparison.

### CODE REVIEW

**Review by jannisborn — APPROVED** [review]
Amazing job @s1k0ra 🤩
